### PR TITLE
Add transform for History

### PR DESCRIPTION
**Pull request by wthrowe**
IMEX preparation.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Time/History.hpp [hunk 1]
```diff
@@ -1008,4 +1008,92 @@ template <typename Vars>
...
+                record.time_step_id, History<DestVars>::no_value,
+                [&](const auto result) {
+                  derivative_transformer(result, record.derivative);
+                });
+          }
+        } else {
+          static_assert(
+              std::is_invocable_v<ValueTransformer, const SourceVars&>,
+              "Transform function must either be callable to mutate entries "
+              "or to generate them.");
```
Comment (nilsdeppe) [inline src/Time/History.hpp]: I found "to generate them" didn't make it obvious that the function should return the transformed state by value. Maybe something like "or return the transformed state by value."?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, one suggestion. Let me know if you agree or not and please squash immediately if you decide to do some sort of reword

File: src/Time/History.hpp (hunk 1 above)
Comment (wthrowe) [reply src/Time/History.hpp]: Done as you suggested.

**Review by nilsdeppe — APPROVED** [review]

**Comment by kidder** [comment]
ignoring known CI issues

**Pull request merged**